automation-for-stable-branches/README.md

```diff
@@ -0,0 +1,66 @@
+# Automation for moving the stable branches
+
+Every Friday we move the stable branches of our repositories so that new `prod` images are rebuilt
+for Packit Service. There are still more and more repositories which
+we need to move the stable branches for and this work is repetitive and can be done automatically.
+
+Repositories which we currently need for Packit Service:
+
+- packit
+- packit-service
+- packit-service-fedmsg
+- packit-service-centosmsg
+- sandcastle
+- dashboard
+- tokman
+
+- ogr (at the moment we have only master branch for it,
+  but we planned to create a stable branch for it as well)
+
+Dependencies (order of moving):
+
+- [packit -> packit-service](https://github.com/packit/packit-service/blob/a1ba5988ce2a2409b4f52f05324587303b52e676/files/install-deps-worker.yaml#L50)
+- [ogr -> packit-service](https://github.com/packit/packit-service/blob/a1ba5988ce2a2409b4f52f05324587303b52e676/files/install-deps-worker.yaml#L52)
+
+AC:
+
+- have a mechanism to determine whether stage works fine
+  and move the stable branches to appropriate commit hashes
+
+## Solution
+
+- create a script which runs the validation of stage and afterwards moving the branches (in one / 2 different scripts)
+- run the script manually/automatically(job)
+- what to do when validation for stage fails
+  - running automatically - notify us - how:
+  - running manually - output failure
+- specific commit-hashes can be defined (so that not the latest commit-hash is used), e.g.:
+  `python3 ./move-branches.py --packit-service=1787687aba71`
+  would move packit-service to the `1787687aba71` and others to `HEAD`
+
+### Determine whether stage works
+Staging app uses images built from master branches, therefore if stage works
+as expected, we should be okay to move all stable branches to the latest commits.
+For this we could use:
+
+- validation script - we could refactor the existing validation script to be able to
+  check stage results instead of prod and test e.g. just one PR used in daily tests,
+  which checks the copr-build and tests triggered by comment (test other triggers?)
+- other options ?
```

Comment:
since the validation script runs on hello-world where we have both stg and prod, we are already checking stg

we just need to figure out how to store the information that the tests passed

Reply:
good point, added